internal/internal_worker_test.go

```diff
@@ -2789,24 +2789,6 @@ func TestWorkerBuildIDAndSessionPanic(t *testing.T) {
 	require.Equal(t, "cannot set both EnableSessionWorker and UseBuildIDForVersioning", recovered)
 }
 
-func TestHistoryFromJSON(t *testing.T) {
```

Comment:
Why are we removing this test?

Reply:
Originally because the logic was moved to api-go, and now because the logic is in temporalproto/decode.go

I can add the test back in here if you want; the decoder test already checks that we can load both history variants

Reply:
Possibly. I'll just add it back to ensure we're covered

Reply:
This will be restored in in my next commit

Reply:
I restored it.